internal/controllers/rest_controller.go

```diff
@@ -126,20 +132,19 @@ func (rc RESTController) OIDCProviderCallback(w http.ResponseWriter, r *http.Req
 	// with the callback from the OIDC Provider.
 	safe, err := rc.oidcService.IsFlowSecure(stateProvider, token)
 	if !safe {
-		rc.logger.Error(err)
 		rc.forbiddenResponse(w, err)
 		return
 	}
 
 	// If the flow is secure, a JWT will be created...
 	jwt, err := rc.oidcService.Create(token.CustomClaims)
 	if err != nil {
-		rc.logger.Errorf("an error occurred while creating a JWT. Error: %v", err)
-		rc.forbiddenResponse(w, err)
+		rc.forbiddenResponse(w, fmt.Errorf("an error occurred while creating a JWT. Error: %v", err))
 		return
 	}
 	// ...set a session cookie.
-	rc.setSessionCookie(w, r, "access_token", jwt.AccessToken)
+	cookieName := rc.credentials["cookie"].GetKeyName()
```

Comment:
Here we can check if the key `cookie` exists before call the method `GetKeyName()`, otherwise, a potential null pointer can happen.

```suggestion
	cookieHandler, ok := rc.credentials["cookie"]
	if !ok {
	   // fail
	}
	cookieName := cookieHandler.GetKeyName()
```